deps/thpool/thpool.c

```diff
@@ -346,25 +456,187 @@ void redisearch_thpool_destroy(redisearch_thpool_t* thpool_p) {
     thread_destroy(thpool_p->threads[n]);
   }
   rm_free(thpool_p->threads);
+  rm_free(thpool_p->name);
   rm_free(thpool_p);
 }
 
 /* Pause all threads in threadpool */
 void redisearch_thpool_pause(redisearch_thpool_t* thpool_p) {
+  if (!thpool_p) return;
+
+  while (thpool_p->flags & RS_THPOOL_F_RESUME) {
+    sleep(1);
+    RedisModule_Log(
+        NULL, "warning",
+        "redisearch_thpool_pause(): waiting for %s threadpool to finish resuming process.",
+        thpool_p->name);
+  }
```

Comment:
We log only if we called pause during resume is progress
also, we wait 1 sec which is equivalent to the time the threads are sleeping when waiting to be resumed, 
so this can only happen if we are trying to pause the threads during a resume is in progress and resuming takes more the 1 sec...a very rare edge case (probably something is wrong)